crates/core/src/client/mod.rs

```diff
@@ -757,4 +777,112 @@ impl<P: Provider + Send + Sync> KakarotStarknetApi<P> for KakarotClient<P> {
+                    entry_point_selector: GET_EVM_ADDRESS,
+                    calldata: vec![],
+                },
+                starknet_block_id,
+            )
+            .await;
+
+        let result: Result<bool, ProviderError<<P as Provider>::Error>> = match result {
+            Ok(_) => Ok(true),
+            Err(error) => match error {
+                ProviderError::StarknetError(error) => match error {
+                    StarknetError::ContractNotFound => Ok(false),
+                    _ => Err(ProviderError::StarknetError(error)),
+                },
+                _ => Err(error),
+            },
+        };
+
+        Ok(result?)
+    }
+
+    async fn deploy_eoa(&self, ethereum_address: Address) -> Result<FieldElement, EthApiError<P::Error>> {
+        let account = &self.starknet_account;
+        let kakarot_address = self.kakarot_address();
+
+        let ethereum_address: FieldElement = Felt252Wrapper::from(ethereum_address).into();
+
+        let result = account
+            .execute(vec![Call {
+                calldata: vec![ethereum_address],
+                to: kakarot_address,
+                selector: DEPLOY_EXTERNALLY_OWNED_ACCOUNT,
+            }])
+            .send()
+            .await;
+
+        let result: Result<FieldElement, EthApiError<P::Error>> = match result {
+            Ok(invoke_result) => Ok(invoke_result.transaction_hash),
+            Err(error) => match error {
+                AccountError::Provider(error) => Err(EthApiError::from(error)),
+                _ => Err(EthApiError::Other(error.into())),
+            },
+        };
+
+        result
+    }
+
+    /// given a transaction hash, waits for it to be confirmed on L2
+    /// each retry is every 1 second, tries for max_retries*1s and then timeouts after that
+    /// returns a boolean indicating whether the transaction was accepted or not L2, it will always
+    /// be true, otherwise should return error
+    async fn wait_for_confirmation_on_l2(
+        &self,
+        transaction_hash: FieldElement,
+        max_retries: u64,
+    ) -> Result<bool, EthApiError<P::Error>> {
+        let mut idx = 0;
+
+        loop {
+            if idx >= max_retries {
+                return Err(EthApiError::Other(anyhow::format_err!(
+                    "timeout: max retries: {max_retries} attempted for polling the transaction receipt"
+                )));
+            }
+
+            let receipt = self.starknet_provider.get_transaction_receipt(transaction_hash).await;
+
+            match receipt {
+                Ok(receipt) => {
+                    if let MaybePendingTransactionReceipt::Receipt(StarknetTransactionReceipt::Invoke(receipt)) =
+                        receipt
+                    {
+                        if let StarknetTransactionStatus::AcceptedOnL2 = receipt.status {
+                            return Ok(true);
+                        };
+                    }
+                }
```

Comment:
I usually prefer to use matches for enums, that way we are sure to handle all variant, while `if let` can make us forget to handle some variants.